### DISCUSSION TITLE
Issue with ejabberd persisting the connection

### DISCUSSION THREAD

**Discussion by mvj2022** [processone/ejabberd#3855] [Questions] [answered]
Hello - 

We are using push notifications to webpage using ejabberd server(version: 16.01-2) and stophe.js. However ejabberd stops sending notifications a short while after establishing the connection. And users have to refresh the page to re-establish the connection. 

It seems that issue may be with ejabberd not being able to persist the connection. Looking at the ejabberd.log, it seems connection is getting terminated every 1min or so. 

Upon looking at error.log, we found following errors logged. I will appreciate, if someone can please help us interpret this line from the log or any other place we should be looking at.

**CRASH REPORT Process <0.29290.5884> with 0 neighbours crashed with reason: no case clause matching {gen_tcp,[<<"XXXX">>,<<"YYYY">>,<<"YYYY]">>]} in ejabberd_http:get_transfer_protocol/2 line 327**

We are also looking at increasing the connection timeout and pinging the server as suggested in other threads. However this issue is happening with only one of the clusters. Other clusters don't have this connection issue. The configuration is the same on both the servers. There we are thinking that the issue may lie somewhere else.

Thanks,

**Comment by prefiks** [comment, marked as answer]
This version is 7 years old, and this bug (i think you are serving from ipv6 address?) was fixed 5 years ago, you need to used never version.

**Reply by mvj2022** [reply]
Thanks prefiks. We will look into it and will update

**Reply by mvj2022** [reply]
I confirmed. We are serving from ipv4 address
